### PR TITLE
EPMRPP-81029 || There is no validation for the creation of dashboards exceeding the limit

### PR DESCRIPTION
**Pull request by APiankouski**

### CODE REVIEW

File: src/main/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImpl.java [hunk 1]
```diff
@@ -121,6 +126,11 @@ public OperationCompletionRS updateDashboard(ReportPortalUser.ProjectDetails pro
 	public OperationCompletionRS addWidget(Long dashboardId, ReportPortalUser.ProjectDetails projectDetails, AddWidgetRq rq,
 			ReportPortalUser user) {
 		Dashboard dashboard = getShareableDashboardHandler.getAdministrated(dashboardId, projectDetails);
+		BusinessRule.expect(dashboard.getDashboardWidgets().size() > MAX_WIDGET_ON_DASHBOARD, Predicates.equalTo(true)).verify(
```
Comment (miracle8484) [inline src/main/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImpl.java:129]: 1. Pleas add test for this point.
2. Please, pay attention to branch naming - it must be with postfix.

File: src/test/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImplTest.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+	@Mock
+	private GetShareableEntityHandler<Dashboard> getShareableDashboardHandler;
+
+	@InjectMocks
+	private UpdateDashboardHandlerImpl updateDashboardHandler;
+
+	private final static int MAX_WIDGET_ON_DASHBOARD = 300;
+
+	@Test
+	void addExistedWidget() {
```
Comment (miracle8484) [inline src/test/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImplTest.java:64]: Please, provide positive test as well.

File: src/main/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImpl.java [hunk 3]
```diff
@@ -126,10 +126,10 @@ public OperationCompletionRS updateDashboard(ReportPortalUser.ProjectDetails pro
 	public OperationCompletionRS addWidget(Long dashboardId, ReportPortalUser.ProjectDetails projectDetails, AddWidgetRq rq,
 			ReportPortalUser user) {
 		Dashboard dashboard = getShareableDashboardHandler.getAdministrated(dashboardId, projectDetails);
-		BusinessRule.expect(dashboard.getDashboardWidgets().size() > MAX_WIDGET_ON_DASHBOARD, Predicates.equalTo(true)).verify(
+		BusinessRule.expect(dashboard.getDashboardWidgets().size() >= MAX_WIDGET_ON_DASHBOARD, Predicates.equalTo(false)).verify(
 				ErrorType.DASHBOARD_UPDATE_ERROR,
 				Suppliers.formattedSupplier(
-						"The limit of 300 dashboards has been reached. To create a new one you need to delete at least one created previously")
+						"The limit of 300 dashboards has been reached. To create a new one you need to delete at least one created previously.")
```
Comment (miracle8484) [inline src/main/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImpl.java:132]: Please, change "300" in error message to placeholder for amount, and fill it with MAX_WIDGET_ON_DASHBOARD.

**Review by miracle8484 — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImplTest.java (hunk 2 above)
Comment (APiankouski) [reply src/test/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImplTest.java:64]: Positive Integration test exist

File: src/main/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImpl.java (hunk 3 above)
Comment (APiankouski) [reply src/main/java/com/epam/ta/reportportal/core/dashboard/impl/UpdateDashboardHandlerImpl.java]: done

**Review by miracle8484 — APPROVED** [review]

**Pull request merged**